packages/components/src/ResourcePicker/Toolbar/NameFilter/NameFilter.component.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import DebounceInput from 'react-debounce-input';
+
+import theme from './NameFilter.scss';
+
+function NameFilter({ label, placeholder, value, debounceTimeout, onChange }) {
+	return (
+		<form className={classNames('name-filter', theme['name-filter'])}>
+			{label && (
+				<label htmlFor="resource-picker-toolbar-name-filter-input" className="sr-only">
+					{label}
+				</label>
+			)}
+			<DebounceInput
+				id="resource-picker-toolbar-name-filter-input"
+				type="text"
+				placeholder={placeholder}
+				value={value}
+				debounceTimeout={debounceTimeout}
+				onChange={onChange}
+				className="form-control"
+				autoComplete="off"
+			/>
+		</form>
+	);
+}
+
+NameFilter.propTypes = {
+	label: PropTypes.string,
+	placeholder: PropTypes.string,
+	value: PropTypes.string,
+	onChange: PropTypes.func,
+	debounceTimeout: PropTypes.number,
```

Comment:
Hmm, not sure that accepting a prop is really relevant, why not use directly 300ms ?